source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -99,6 +109,14 @@ MGTransferMatrixFree<dim, Number>::build(
   const std::vector<std::shared_ptr<const Utilities::MPI::Partitioner>>
     &external_partitioners)
 {
+  if (external_partitioners.size() > 0)
+    this->initialize_dof_vector =
+      [&external_partitioners](
+        const unsigned int                          level,
+        LinearAlgebra::distributed::Vector<Number> &vec) {
+        vec.reinit(external_partitioners[level]);
+      };
```

Comment:
@kronbichler I have rebased this PR and have converted the vector of partitioners internally to a lambda function. What do you think. As far as I see it, the approach with `adjust_ghost_range_if_necessary()` is broken, since `Multigrid` will reset the partitioners in each cycle and I don't see a way to fix this. We might also want to consider to update step-59.

Reply:
Do you mean that it calls `adjust_ghost_range_if_necessary` exactly once per level in *each* multigrid iteration (as the vector is reset upon `copy_to_mg`)? Either way, as discussed in the post below, I think we should improve the initialization of all vectors to avoid these kind of operations.

Reply:
> Do you mean that it calls adjust_ghost_range_if_necessary exactly once per level in each multigrid iteration (as the vector is reset upon copy_to_mg)?

Yes. That is the case.